cloud/linode/loadbalancers.go

```diff
@@ -178,31 +178,32 @@ func (l *loadbalancers) UpdateLoadBalancer(ctx context.Context, clusterName stri
 		}
 
 		// Look for an existing config for this port
-		var existingNBCfg *linodego.NodeBalancerConfig
-		for _, nbc := range nbCfgs {
-			nbc := nbc
+		var currentNBCfg *linodego.NodeBalancerConfig
+		for i := range nbCfgs {
+			nbc := nbCfgs[i]
 			if nbc.Port == int(port.Port) {
-				existingNBCfg = &nbc
+				currentNBCfg = &nbc
 				break
 			}
 		}
 
-		// If there's an existing config, rebuild it, otherwise, create it
-		if existingNBCfg != nil {
-			rebuildOpts := newNBCfg.GetRebuildOptions()
-			rebuildOpts.Nodes = newNBNodes
-
-			if _, err = l.client.RebuildNodeBalancerConfig(ctx, lb.ID, existingNBCfg.ID, rebuildOpts); err != nil {
-				return fmt.Errorf("[port %d] error rebuilding NodeBalancer config: %v", int(port.Port), err)
-			}
-		} else {
+		// If there's no existing config, create it
+		if currentNBCfg == nil {
 			createOpts := newNBCfg.GetCreateOptions()
-			createOpts.Nodes = newNBNodes
 
-			_, err := l.client.CreateNodeBalancerConfig(ctx, lb.ID, createOpts)
+			newNBCfg, err := l.client.CreateNodeBalancerConfig(ctx, lb.ID, createOpts)
```

Comment:
I believe the `err` is defined already, so you can do `currentNBCfg, err = l.client...` to save two lines below